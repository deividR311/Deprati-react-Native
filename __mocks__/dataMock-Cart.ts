export const DataMock_Cart = {
  carts: [
    {
      appliedOrderPromotions: [],
      appliedProductPromotions: [],
      appliedVouchers: [],
      calculated: true,
      code: '4809002',
      deliveryAddress: {
        cellphoneNumber: '34949494',
        cellphonePreffix: '09',
        city: 'COT-287',
        country: {
          isocode: 'EC',
          name: 'Ecuador'
        },
        defaultAddress: true,
        email: 'alexis.enriquezca@gmail.com',
        firstName: 'Alejandra ',
        formattedAddress: 'Calle 6, Casa 23, Cotopaxi, Alaquez',
        id: '8819555172375',
        idNumber: '1721480299',
        idType: 'CEDULA',
        line1: 'Calle 6',
        line2: 'Casa 23',
        otherInfo: 'Laura',
        phone: '',
        region: {
          countryIso: 'EC',
          isocode: 'EC-COT',
          isocodeShort: '193',
          name: 'Cotopaxi'
        },
        shippingAddress: true,
        town: 'Alaquez',
        visibleInAddressBook: true
      },
      deliveryItemsQuantity: 1,
      deliveryOrderGroups: [
        {
          entries: [
            {
              basePrice: {
                currencyIso: 'USD',
                formattedValue: '$2,32',
                priceType: 'BUY',
                value: 2.32
              },
              basePriceWithTaxes: {
                currencyIso: 'USD',
                formattedValue: '$2,60',
                priceType: 'BUY',
                value: 2.5984
              },
              basePriceWithoutTaxes: {
                currencyIso: 'USD',
                formattedValue: '$23,20',
                priceType: 'BUY',
                value: 23.2
              },
              deliveryMode: {
                code: 'standard'
              },
              deliveryTimeRange: '5 a 7 días laborables',
              entryNumber: 0,
              giftPacking: false,
              pickup: false,
              product: {
                acceptGiftPacking: true,
                availableForPickup: false,
                baseOptions: [
                  {
                    selected: {
                      code: '226010190802030008',
                      priceData: {
                        currencyIso: 'USD',
                        formattedValue: '$2,32',
                        priceType: 'BUY',
                        value: 2.32
                      },
                      stock: {
                        stockLevel: 11,
                        stockLevelStatus: 'inStock'
                      },
                      url: '/camiseta-estampada/p/226010190802030008',
                      variantOptionQualifiers: []
                    },
                    variantType: 'GenericVariantProduct'
                  }
                ],
                baseProduct: '000226010190802030',
                categories: [
                  {
                    code: 'T8023',
                    name: '10',
                    parentCategoryName: 'Talla',
                    url: '/10/c/T8023'
                  },
                  {
                    code: 'C0110601',
                    name: 'Blanco',
                    parentCategoryName: 'Color',
                    url: '/blanco/c/C0110601'
                  },
                  {
                    code: '0701',
                    name: 'MUJERES',
                    url: '/mujeres/c/0701'
                  }
                ],
                code: '226010190802030008',
                deliveryModes: [
                  {
                    code: 'standard'
                  },
                  {
                    code: 'pickup'
                  }
                ],
                ean: '15282579',
                images: [
                  {
                    format: 'zoom',
                    imageType: 'PRIMARY',
                    url: 'https://imagestg.deprati.com.ec/sys-master/images/h67/h07/8890805157918/15282579-0_product_515Wx772H'
                  },
                  {
                    format: 'product',
                    imageType: 'PRIMARY',
                    url: 'https://imagestg.deprati.com.ec/sys-master/images/h26/h0b/8890805288990/15282579-0_product_300Wx450H'
                  },
                  {
                    format: 'thumbnail',
                    imageType: 'PRIMARY',
                    url: 'https://imagestg.deprati.com.ec/sys-master/images/h38/h0e/8890805354526/15282579-0_product_96Wx144H'
                  },
                  {
                    format: 'cartIcon',
                    imageType: 'PRIMARY',
                    url: 'https://imagestg.deprati.com.ec/sys-master/images/h49/h11/8890805420062/15282579-0_product_65Wx97H'
                  }
                ],
                isGiftProduct: false,
                isPromotionSpecialPrice: false,
                ivaCondition: '12%',
                name: 'CAMISETA ESTAMPADA',
                purchasable: true,
                stock: {
                  stockLevel: 11,
                  stockLevelStatus: 'inStock'
                },
                url: '/camiseta-estampada/p/226010190802030008'
              },
              quantity: 1,
              subtotal: {
                currencyIso: 'USD',
                formattedValue: '$23,20',
                priceType: 'BUY',
                value: 23.2
              },
              totalDiscount: {
                currencyIso: 'USD',
                formattedValue: '$20,88',
                priceType: 'BUY',
                value: 20.88
              },
              totalPrice: {
                currencyIso: 'USD',
                formattedValue: '$2,32',
                priceType: 'BUY',
                value: 2.32
              },
              totalPriceWithoutTaxes: {
                currencyIso: 'USD',
                formattedValue: '$2,32',
                priceType: 'BUY',
                value: 2.32
              },
              updateable: true,
              volume: 0,
              weight: 0
            }
          ],
          totalPriceWithTax: {
            currencyIso: 'USD',
            formattedValue: '$2,60',
            priceType: 'BUY',
            value: 2.5999999999999996
          }
        }
      ],
      entries: [
        {
          basePrice: {
            currencyIso: 'USD',
            formattedValue: '$2,32',
            priceType: 'BUY',
            value: 2.32
          },
          basePriceWithTaxes: {
            currencyIso: 'USD',
            formattedValue: '$2,60',
            priceType: 'BUY',
            value: 2.5984
          },
          basePriceWithoutTaxes: {
            currencyIso: 'USD',
            formattedValue: '$23,20',
            priceType: 'BUY',
            value: 23.2
          },
          deliveryMode: {
            code: 'standard'
          },
          deliveryTimeRange: '5 a 7 días laborables',
          entryNumber: 0,
          giftPacking: false,
          pickup: false,
          product: {
            acceptGiftPacking: true,
            availableForPickup: false,
            baseOptions: [
              {
                selected: {
                  code: '226010190802030008',
                  priceData: {
                    currencyIso: 'USD',
                    formattedValue: '$2,32',
                    priceType: 'BUY',
                    value: 2.32
                  },
                  stock: {
                    stockLevel: 11,
                    stockLevelStatus: 'inStock'
                  },
                  url: '/camiseta-estampada/p/226010190802030008',
                  variantOptionQualifiers: []
                },
                variantType: 'GenericVariantProduct'
              }
            ],
            baseProduct: '000226010190802030',
            categories: [
              {
                code: 'T8023',
                name: '10',
                parentCategoryName: 'Talla',
                url: '/10/c/T8023'
              },
              {
                code: 'C0110601',
                name: 'Blanco',
                parentCategoryName: 'Color',
                url: '/blanco/c/C0110601'
              },
              {
                code: '0701',
                name: 'MUJERES',
                url: '/mujeres/c/0701'
              }
            ],
            code: '226010190802030008',
            deliveryModes: [
              {
                code: 'standard'
              },
              {
                code: 'pickup'
              }
            ],
            ean: '15282579',
            images: [
              {
                format: 'zoom',
                imageType: 'PRIMARY',
                url: 'https://imagestg.deprati.com.ec/sys-master/images/h67/h07/8890805157918/15282579-0_product_515Wx772H'
              },
              {
                format: 'product',
                imageType: 'PRIMARY',
                url: 'https://imagestg.deprati.com.ec/sys-master/images/h26/h0b/8890805288990/15282579-0_product_300Wx450H'
              },
              {
                format: 'thumbnail',
                imageType: 'PRIMARY',
                url: 'https://imagestg.deprati.com.ec/sys-master/images/h38/h0e/8890805354526/15282579-0_product_96Wx144H'
              },
              {
                format: 'cartIcon',
                imageType: 'PRIMARY',
                url: 'https://imagestg.deprati.com.ec/sys-master/images/h49/h11/8890805420062/15282579-0_product_65Wx97H'
              }
            ],
            isGiftProduct: false,
            isPromotionSpecialPrice: false,
            ivaCondition: '12%',
            name: 'CAMISETA ESTAMPADA',
            purchasable: true,
            stock: {
              stockLevel: 11,
              stockLevelStatus: 'inStock'
            },
            url: '/camiseta-estampada/p/226010190802030008'
          },
          quantity: 1,
          subtotal: {
            currencyIso: 'USD',
            formattedValue: '$23,20',
            priceType: 'BUY',
            value: 23.2
          },
          totalDiscount: {
            currencyIso: 'USD',
            formattedValue: '$20,88',
            priceType: 'BUY',
            value: 20.88
          },
          totalPrice: {
            currencyIso: 'USD',
            formattedValue: '$2,32',
            priceType: 'BUY',
            value: 2.32
          },
          totalPriceWithoutTaxes: {
            currencyIso: 'USD',
            formattedValue: '$2,32',
            priceType: 'BUY',
            value: 2.32
          },
          updateable: true,
          volume: 0,
          weight: 0
        }
      ],
      guid: '15901dbd-9966-4239-9a4d-fecc4f747589',
      net: true,
      orderDiscounts: {
        currencyIso: 'USD',
        formattedValue: '$0,00',
        priceType: 'BUY',
        value: 0
      },
      pickupItemsQuantity: 0,
      pickupOrderGroups: [],
      productDiscounts: {
        currencyIso: 'USD',
        formattedValue: '$0,00',
        priceType: 'BUY',
        value: 0
      },
      sellWithTaxes: {
        currencyIso: 'USD',
        formattedValue: '$2,32',
        priceType: 'BUY',
        value: 2.32
      },
      sellWithoutTaxWithoutInterest: {
        currencyIso: 'USD',
        formattedValue: '$2,31',
        priceType: 'BUY',
        value: 2.31
      },
      sellWithoutTaxes: {
        currencyIso: 'USD',
        formattedValue: '$0,00',
        priceType: 'BUY',
        value: 0
      },
      site: 'deprati',
      store: 'deprati',
      subTotal: {
        currencyIso: 'USD',
        formattedValue: '$2,32',
        priceType: 'BUY',
        value: 2.32
      },
      totalDiscounts: {
        currencyIso: 'USD',
        formattedValue: '$20,88',
        priceType: 'BUY',
        value: 20.88
      },
      totalItems: 1,
      totalPrice: {
        currencyIso: 'USD',
        formattedValue: '$2,32',
        priceType: 'BUY',
        value: 2.32
      },
      totalPriceWithTax: {
        currencyIso: 'USD',
        formattedValue: '$2,60',
        priceType: 'BUY',
        value: 2.6
      },
      totalTax: {
        currencyIso: 'USD',
        formattedValue: '$0,28',
        priceType: 'BUY',
        value: 0.28
      },
      totalVolume: 0,
      totalWeight: 0.25,
      user: {
        name: 'Alexis pruebas ENRIQUEZ',
        uid: 'alexis.enriquezca@gmail.com'
      },
      potentialOrderPromotions: [
        {
          consumedEntries: [],
          promotion: {
            code: '10012022_EGCE_PRUEBAAPP1_A',
            description: 'Version 2 Hybris - PRUEBA APP',
            endDate: '2024-01-01T04:59:59+0000',
            promotionType: 'Promoción basada en regla'
          }
        },
        {
          consumedEntries: [],
          description:
            'Completa $150 y  obten envío gratis. Aplica restricciones.',
          promotion: {
            code: '10012022_EGCE_PRUEBAAPP1_C',
            description: 'Version 2 Hybris',
            endDate: '2024-01-01T04:59:59+0000',
            promotionType: 'Promoción basada en regla'
          }
        }
      ],
      potentialProductPromotions: [],
      totalUnitCount: 1
    }
  ]
};
