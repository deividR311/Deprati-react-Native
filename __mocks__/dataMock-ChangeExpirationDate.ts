export const DataMock_ChangeExpirationDate = {
  TEXT_NEWDATE: 'Selecciona el día',
  infoCreditBalance: {
    clientName: 'PATRICIA L. DE UGALDE',
    cardNumber: '1xxxxx6',
    affiliateDate: '2000',
    totalAmount: '5.500,00',
    amountSpent: '724,35',
    aditional: '00',
    availableAmount: '4.775,65',
    pastDueDays: 0,
    nextPayDay: 'Abril 14 de 2023',
    nextAmountToPay: '105,08',
    numeroTarjetaAdicionalDisplay: '14XXX86 00',
    clientFirstName: 'Prueba'
  },
  dataCurrentDate: {
    cuttingDay: 14,
    dueDate: '20230514',
    day: '14'
  },
  dataChangeDate: null,
  selectedDay: '2',
  isAcceptTerms: false,
  showWarning: false,
  isLoadingChangeDate: false,
  isLoadingCurrentDate: false,
  showSucces: false,
  showErrorSucces: false,
  expirationDateContent: {
    legend:
      'No puedes cambiar tu fecha máxima de pago si tienes una deuda pendiente y <b>este cambio se hará efectivo a partir del próximo mes.</b>',
    popup_title:
      '¿Estás seguro de cambiar la fecha máxima de pago de tu crédito De Prati?',
    popup_paragraphs: [
      'Ten en cuenta:',
      '1. Esta acción no se puede deshacer y el cambio se hará efectivo a partir de tu próximo mes.',
      '2. Si haces el cambio de fecha <b>deberás pagar la cuota del mes en curso</b> y este cambio se hará efectivo a partir del próximo mes.'
    ],
    popup_accept_action: 'SI, ESTOY DE ACUERDO',
    popup_cancel_action: 'NO, CANCELAR'
  },
  selectRef: {
    current: {
      reset: jest.fn(),
      openDropdown: jest.fn(),
      closeDropdown: jest.fn()
    }
  },
  setSelectedDay: jest.fn(),
  setIsAcceptTerms: jest.fn(),
  handleChangeDate: jest.fn(),
  setShowWarning: jest.fn(),
  handleCloseSuccess: jest.fn(),
  handleCloseError: jest.fn(),
  handleCloseWarning: jest.fn()
};
