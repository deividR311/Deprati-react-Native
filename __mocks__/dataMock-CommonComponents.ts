export const DataMock_CustomProps = {
  url: 'url',
  style: {},
  content: 'qwerty',
  uid: 'uid',
  typeCode: 'qwerty',
  modifiedTime: 'qwerty',
  name: 'qwerty',
  container: true,
  rotatingImagesComponentsList: 'DepratiBrandsRotatingImages',
  childrenComponentsList: ['DepratiBrandsRotatingImages'],
  childrenComponents: {
    DepratiBrandsRotatingImages: {
      uid: 'DepratiBrandsRotatingImages',
      typeCode: 'RotatingImagesComponent',
      modifiedTime: '2023-04-28T12:59:58+0000',
      name: 'DePrati Dior Mobile Carousel',
      container: 'false',
      resolutionEnum: 'MOBILE',
      effect: 'FADE',
      banners: 'DiorBannerComponent2 BannerCarouselDiorVideo_1',
      timeout: '6',
      childrenComponentsList: [
        'DiorBannerComponent2',
        'BannerCarouselDiorVideo_1'
      ]
    }
  }
};

export const DataMock_BillingAddressComponent = {
  style: {},
  onSelected: () => {},
  emptyStyle: {},
  showTitle: true,
  enableHandleEnableButton: true,
  onIsEnable: () => {},
  paymentAddress: {
    defaultAddress: false,
    email: 'string',
    firstName: 'string',
    formattedAddress: 'string',
    id: 'string',
    line1: 'string',
    line2: 'string',
    phone: 'string',
    phonePreffix: 'string',
    shippingAddress: false,
    visibleInAddressBook: false
  },
  paymentCredit: {
    showAlerAddressPayment: false,
    setShowAlerAddressPayment: () => {}
  },
  styleSecundaryButton: false
};

export const DataMock_useBillingAddress = {
  addressSelected: {
    defaultAddress: false,
    firstName: 'firstName',
    id: 'qwertystring',
    line1: 'qwerty',
    line2: 'qwerty',
    phone: 'qwerty',
    phonePreffix: 'qwerty'
  },
  isLoadingByDeleting: false,
  addressPaymentList: {
    default: null,
    others: [],
    existsAddress: false,
    loading: false
  },
  isLoading: false,
  buttonChangeData: false,
  handleChangeData: jest.fn(),
  changeAddress: {
    handleSelected: jest.fn(),
    addressSelected: undefined
  },
  handleDelete: jest.fn(),
  handleAddEdit: jest.fn(),
  handleBillingAddressList: () => {
    return [
      {
        defaultAddress: false,
        email: 'string',
        firstName: 'string',
        formattedAddress: 'string',
        id: 'string',
        line1: 'string',
        line2: 'string',
        phone: 'string',
        phonePreffix: 'string',
        shippingAddress: false,
        visibleInAddressBook: false
      }
    ];
  }
};

export const DataMock_IncreaseCreditLimit = {
  rectangularSize: 232,
  value: 1000,
  maxValue: 2000,
  onChangeCreditLimit: jest.fn()
};

export const DataMock_Charts = {
  constentChart: {
    availableAmount: 100,
    amountSpent: 20,
    totalAmount: 200
  },
  ownerFullname: 'Custom customer customization',
  invertValues: true,
  showDisplayCard: true
};
