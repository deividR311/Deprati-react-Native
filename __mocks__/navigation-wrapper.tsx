import React from 'react';
import { NavigationContainer } from '@react-navigation/native';
import { createNativeStackNavigator } from '@react-navigation/native-stack';
import { COLORS } from '../src/application/common';
import { ProvideWrapper } from './provider-wrapper';

jest.useFakeTimers();
jest.mock('react-native/Libraries/Animated/NativeAnimatedHelper');

const Stack = createNativeStackNavigator();

export const NavigationWrapper = ({ children, screen, initialParams }: any) => (
  <ProvideWrapper>
    <NavigationContainer>
      <Stack.Navigator
        initialRouteName="default"
        screenOptions={{
          headerShown: true,
          headerShadowVisible: false,
          headerTitleAlign: 'left',
          contentStyle: { backgroundColor: COLORS.WHITE }
        }}>
        {screen && (
          <Stack.Screen
            name="default"
            initialParams={initialParams ?? {}}
            component={screen}
          />
        )}
        {children && <Stack.Screen name="default" component={() => children} />}
      </Stack.Navigator>
    </NavigationContainer>
  </ProvideWrapper>
);
