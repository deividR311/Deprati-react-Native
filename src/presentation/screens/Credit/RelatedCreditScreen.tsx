import React, { useLayoutEffect } from 'react';
import { Text, View } from 'react-native';
import { COLORS } from '../../../application/common/colors';
import useGetCreditsUser from '../../../application/common/hooksCommons/useGetCreditsUser';
import { NAV } from '../../../application/common/namesNav';
import { LocalStorageKey } from '../../../application/state-manager/services/localstorage';
import { useLocalStorage } from '../../../application/state-manager/services/localstorage/useLocalStorage';
import { ButtonText } from '../../common-components/buttons/Button';
import ImageCard from '../../common-components/credito/ImageCard';
import { stylesSignUp } from '../SignUp/stylesSignUp';
import ButtonRelatedCredit from './components/ButtonRelatedCredit';
import { StackActions, useNavigation } from '@react-navigation/native';
import { useTranslation } from 'react-i18next';

export default function RelatedCreditScreen() {
  const { t } = useTranslation();
  const navigation = useNavigation();
  const { save, localStorageData } = useLocalStorage();
  const { isError, isLoading, loadCreditsUser } = useGetCreditsUser();

  const handleOmitir = () => {
    save({
      [LocalStorageKey.IsSucessFirstLogin]: true
    });
    navigation.dispatch(StackActions.replace(NAV.DASHBOARD_NAVIGATION));
  };

  useLayoutEffect(() => {
    loadCreditsUser();
  }, [localStorageData[LocalStorageKey.User]]);

  return (
    <View style={{ marginHorizontal: 16, flex: 1 }}>
      <ImageCard />
      <Text style={stylesSignUp.titleCredit}> {t('creditSaveTime')} </Text>
      <Text style={stylesSignUp.subtitle1}>{t('creditLink1')} &nbsp;</Text>
      <ButtonRelatedCredit
        disabled={isError}
        title={'Vincular crÃ©dito de Prati'.toLocaleUpperCase()}
      />
      <View style={{ flex: 1, height: '100%' }} />
      <ButtonText
        testID={'vincular_credito_omitir'}
        style={{
          marginBottom: 16
        }}
        styleText={{ color: COLORS.BRAND }}
        title="Omitir"
        onPress={() => {
          handleOmitir();
        }}
      />
    </View>
  );
}
