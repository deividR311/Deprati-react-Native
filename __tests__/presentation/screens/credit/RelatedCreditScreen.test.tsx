import React from 'react';
import { render } from '@testing-library/react-native';
import { NavigationWrapper } from '../../../../__mocks__/navigation-wrapper';
import RelatedCreditScreen from '../../../../src/presentation/screens/Credit/RelatedCreditScreen';

describe('RelatedCreditScreen', () => {
  it('renders correctly in Snapshot', () => {
    const screenTest = render(
      <NavigationWrapper screen={RelatedCreditScreen} />
    );
    expect(screenTest?.toJSON()).toMatchSnapshot();
  });
});
