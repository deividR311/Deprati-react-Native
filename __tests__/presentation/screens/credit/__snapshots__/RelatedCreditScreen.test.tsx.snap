// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RelatedCreditScreen renders correctly in Snapshot 1`] = `
<RNCSafeAreaProvider
  onInsetsChange={[Function]}
  style={
    Array [
      Object {
        "flex": 1,
      },
      undefined,
    ]
  }
>
  <RNSScreenStack
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <RNSScreen
      collapsable={false}
      forwardedRef={[Function]}
      gestureResponseDistance={
        Object {
          "bottom": -1,
          "end": -1,
          "start": -1,
          "top": -1,
        }
      }
      onAppear={[Function]}
      onDisappear={[Function]}
      onDismissed={[Function]}
      onTransitionProgress={[Function]}
      onWillDisappear={[Function]}
      replaceAnimation="push"
      stackPresentation="push"
      style={
        Object {
          "bottom": 0,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        }
      }
      swipeDirection="horizontal"
    >
      <RNSScreenStackHeaderConfig
        backButtonInCustomView={false}
        backgroundColor="rgb(255, 255, 255)"
        color="rgb(0, 122, 255)"
        direction="ltr"
        disableBackButtonMenu={false}
        hidden={false}
        hideBackButton={false}
        hideShadow={true}
        largeTitleHideShadow={false}
        title="default"
        titleColor="rgb(28, 28, 30)"
        topInsetEnabled={false}
        translucent={false}
      />
      <View
        style={
          Array [
            Object {
              "flex": 1,
            },
            Object {
              "backgroundColor": "rgb(242, 242, 242)",
            },
            Object {
              "backgroundColor": "#ffffff",
            },
          ]
        }
      >
        <View
          style={
            Object {
              "flex": 1,
              "marginHorizontal": 16,
            }
          }
        >
          <View
            accessibilityIgnoresInvertColors={true}
            style={
              Array [
                Object {
                  "alignSelf": "center",
                  "height": 143,
                  "justifyContent": "center",
                  "marginTop": 32,
                  "width": 235,
                },
                undefined,
              ]
            }
          >
            <Image
              resizeMode="contain"
              source={
                Object {
                  "testUri": "../../../assets/images/cardCredito.png",
                }
              }
              style={
                Array [
                  Object {
                    "bottom": 0,
                    "left": 0,
                    "position": "absolute",
                    "right": 0,
                    "top": 0,
                  },
                  Object {
                    "height": 143,
                    "width": 235,
                  },
                  undefined,
                ]
              }
            />
          </View>
          <Text
            style={
              Object {
                "fontFamily": "Roboto-Bold",
                "fontSize": 16,
                "marginBottom": 16,
                "marginTop": 24,
                "textAlign": "center",
              }
            }
          >
             
            creditSaveTime
             
          </Text>
          <Text
            style={
              Object {
                "alignSelf": "center",
                "color": "#000000",
                "fontFamily": "Roboto-Regular",
                "fontSize": 16,
                "lineHeight": 24,
                "marginBottom": 24,
                "textAlign": "center",
                "width": "100%",
              }
            }
          >
            creditLink1
              
          </Text>
          <View
            accessibilityLabel=""
            accessibilityState={
              Object {
                "disabled": false,
              }
            }
            accessible={true}
            collapsable={false}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "#E31A2C",
                "borderRadius": 5,
                "justifyContent": "center",
                "marginVertical": 16,
                "opacity": 1,
                "paddingVertical": 10,
                "width": "100%",
              }
            }
            testID=""
          >
            <View
              style={
                Object {
                  "flexDirection": "row",
                  "justifyContent": "space-between",
                }
              }
            >
              <Text
                style={
                  Array [
                    Object {
                      "color": "white",
                      "fontFamily": undefined,
                      "fontSize": 14,
                      "fontStyle": "normal",
                      "fontWeight": undefined,
                    },
                    Object {},
                    Object {},
                  ]
                }
              >
                INICIAR SESIÓN
              </Text>
            </View>
          </View>
          <Modal
            animationType="fade"
            hardwareAccelerated={false}
            onDismiss={[Function]}
            transparent={true}
            visible={false}
          />
          <Modal
            animationType="fade"
            hardwareAccelerated={false}
            onDismiss={[Function]}
            transparent={true}
            visible={false}
          />
          <View
            style={
              Object {
                "flex": 1,
                "height": "100%",
              }
            }
          />
          <View
            accessibilityLabel="vincular_credito_omitir"
            accessibilityState={
              Object {
                "disabled": false,
              }
            }
            accessible={true}
            collapsable={false}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "alignItems": "center",
                "justifyContent": "center",
                "marginBottom": 16,
                "opacity": 1,
                "paddingVertical": 14,
              }
            }
            testID="vincular_credito_omitir"
          >
            <Text
              style={
                Array [
                  Object {
                    "color": "#263238",
                    "fontFamily": "Roboto-Regular",
                    "fontSize": 14,
                    "fontWeight": undefined,
                    "letterSpacing": 0.6,
                    "textDecorationLine": "underline",
                  },
                  Object {
                    "color": "#E31A2C",
                  },
                  Object {},
                ]
              }
            >
              Omitir
            </Text>
          </View>
        </View>
      </View>
    </RNSScreen>
  </RNSScreenStack>
</RNCSafeAreaProvider>
`;
